<launch>
  <arg name="compressed" default="true" />
  <arg name="retina_mask" default="false" />
  <arg name="agnostic_nms" default="false" />
  <arg name="conf" default="0.4" />
  <arg name="iou" default="0.8" />
  <arg name="image_topic" default="/camera/color/image_raw"/>

  <node pkg="fastsam_trt_ros" name="fastsam_ros" type="fastsam_ros.py" output="screen">
    <!-- <param name="image_in" value="/zed2i/zed_node/rgb/image_rect_color"/> -->
    <remap from="/image_in" to="$(arg image_topic)"/>
    <remap from="/image_in/compressed" to="$(arg image_topic)/compressed"/>
    <remap from="/detections" to="/detector/detections"/>
    <!-- <remap from="~image_out" to="/detector_out/compressed"/> -->
    <param name="model_path" value="$(find fastsam_trt_ros)/models/fast_sam_x_fp16_opt_threads_480.trt" />
    <param name="compressed" value="$(arg compressed)" />
    <param name="conf" value="$(arg conf)" />
    <param name="iou" value="$(arg iou)" />
    <param name="retina_mask" value="$(arg retina_mask)" />
    <param name="agnostic_nms" value="$(arg agnostic_nms)" />
    <param name="image_size" value="480"/> 
  </node>

  <node pkg="yolov7_trt_ros" name="detector" type="detect_ros.py" output="screen">
    <!-- <param name="image_in" value="/zed2i/zed_node/rgb/image_rect_color"/> -->
    <param name="image_in" value="$(arg image_topic)"/>
    <param name="image_out" value="/detections"/>
    <param name="model_path" value="$(find yolov7_trt_ros)/models/yolov7v42-tiny.trt" />
    <param name="compressed" value="$(arg compressed)" />
    <!-- <param name="topic_in" value="$(arg topic_name)"/> -->
    <param name="display" value="true"/>
    <param name="threshold" value="$(arg conf)" />
  </node>

</launch>